File: forge-cli/pkg/profile/profiles/pytest/templates/test_file.py
"""E2E test template for pytest.

Place in tests/e2e/features/<slug>/test_<feature>.py
Run with: python -m pytest tests/e2e/features/<slug>/ -v
"""

import pytest
import subprocess
import os


def test_tc_nnn_description():
    """Traceability: TC-NNN → {PRD Source}"""
    # Step 1: Setup
    # Step 2: Execute
    result = subprocess.run(
        ["binary", "--flag", "value"],
        capture_output=True, text=True
    )

    # Expected: ...
    assert result.returncode == 0, f"command failed: {result.stderr}"
    assert "expected text" in result.stdout

File: forge-cli/pkg/profile/languages/rust/templates/test_file.rs
/// E2E test template for Rust.
/// Place in tests/e2e/features/<slug>/e2e_tests.rs
/// Build tag: run with `cargo test --test e2e`

use std::process::Command;

/// Traceability: TC-NNN → {PRD Source}
#[test]
fn test_tc_nnn_description() {
    // Step 1: Setup
    // Step 2: Execute
    let output = Command::new("./target/debug/binary")
        .args(&["--flag", "value"])
        .output()
        .expect("failed to execute binary");

    // Expected: ...
    assert!(output.status.success(), "command failed: {}", String::from_utf8_lossy(&output.stderr));
    assert!(String::from_utf8_lossy(&output.stdout).contains("expected text"));
}
